guides/bug_report_templates/action_text_gem.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require "bundler/inline"
+
+gemfile(true) do
+  source "https://rubygems.org"
+
+  git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+
+  gem "capybara"
+  gem "rails", "~> 7.1.0"
+  gem "selenium-webdriver", "~> 4.11.0"
+  gem "sqlite3"
+end
+require "action_controller/railtie"
+require "active_record/railtie"
+require "active_storage/engine"
+require "action_text/engine"
+
+class TestApp < Rails::Application
+  config.load_defaults Rails::VERSION::STRING.to_f
+
+  config.root = __dir__
+  config.hosts << "example.org"
+  config.eager_load = false
+  config.session_store :cookie_store, key: "cookie_store_key"
+  config.secret_key_base = "secret_key_base"
+
+  config.logger = Logger.new($stdout)
+  Rails.logger  = config.logger
+
+  config.active_storage.service = :local
+  config.active_storage.service_configurations = {
+    local: {
+      root: Dir.tmpdir,
+      service: "Disk"
+    }
+  }
+
+  routes.draw do
+    resources :articles, only: [:new, :create, :show]
+  end
+end
+
+ENV["DATABASE_URL"] = "sqlite3::memory:"
+
+Rails.application.initialize!
+
+require ActiveStorage::Engine.root.join("db/migrate/20170806125915_create_active_storage_tables.rb").to_s
+require ActionText::Engine.root.join("db/migrate/20180528164100_create_action_text_tables.rb").to_s
+
+ActiveRecord::Schema.define do
+  CreateActiveStorageTables.new.change
+  CreateActionTextTables.new.change
+
+  create_table :articles, force: true
+end
+
+class Article < ActiveRecord::Base
+  has_rich_text :body
+end
+
+class ArticlesController < ActionController::Base
+  def new
+    @article = Article.new
+
+    render inline: <<~ERB
+      <%= form_with model: @article %>
+        <%= form.rich_text_area :body %>
+        <%= form.button %>
+      <% end %>
+    ERB
+  end
+
+  def create
+    @article = Article.create!(params.require(:article).permit(:body))
+
+    redirect_to @article
+  end
+
+  def show
+    @article = Article.find(params[:id])
+
+    render inline: "<%= @article.body %>"
+  end
+end
+
+require "minitest/autorun"
+
+Capybara.server = :puma, { Silent: true }
+
+class BugTest < ActionDispatch::SystemTestCase
+  test "Action Text" do
+    visit new_articles_path
```

Comment:
@p8 I'm not sure why, but the routing helpers aren't available. What configuration step is this file missing?

Reply:
@rafaelfranca are there any configuration steps that might be missing in this file?

Reply:
Not sure on solution. But the issue is not with the routing helper. The route itself isn't being defined.

If you add this:

```ruby
routes = Rails.application.routes.routes
paths = routes.collect {|r| r.path.spec.to_s }
puts paths
```

After `Rails.application.initialize!` you get this:

```
/rails/info/properties(.:format)
/rails/info/routes(.:format)
/rails/info(.:format)
/rails/active_storage/blobs/redirect/:signed_id/*filename(.:format)
/rails/active_storage/blobs/proxy/:signed_id/*filename(.:format)
/rails/active_storage/blobs/:signed_id/*filename(.:format)
/rails/active_storage/representations/redirect/:signed_blob_id/:variation_key/*filename(.:format)
/rails/active_storage/representations/proxy/:signed_blob_id/:variation_key/*filename(.:format)
/rails/active_storage/representations/:signed_blob_id/:variation_key/*filename(.:format)
/rails/active_storage/disk/:encoded_key/*filename(.:format)
/rails/active_storage/disk/:encoded_token(.:format)
/rails/active_storage/direct_uploads(.:format)
```

So, the routes defined in this test aren't showing there. Not sure why yet.

Reply:
You have to use `append`, otherwise the routes get overriden... I wrote a bit about this here: https://github.com/jeremyevans/r10k/commit/fd86a4e8b50e3e1f3924f82170f90336ce48a1ce

> My understanding is that Rails is hard coded during initialization to
call #clear! on the RouteSet and then always try to load
config/routes.rb. Therefore, we cannot use #draw because whatever we
draw will be cleared, whereas appended routes are not cleared by #clear!
While this isn't "default" Rails configuration, it seems like a
reasonable solution for a single file Rails application that results in
more accurate behavior.

Reply:
I have tried calling `append`, both before and after the call to `Rails.application.initialize!`, but neither have any effect -- the test still fails with a `NoMethodError`.

I've also replaced `new_article_path` with `"/articles/new"` in an attempt to tell whether or not the issue was the helpers themselves or the underlying routes. The tests fail because `"/articles/new"` isn't recognized.

Reply:
@skipkayhil thank you! I was so set on debugging the routing that I didn't read the error message citing the route that legitimately did not exist. I've resolved the routing issues, but there are new failures! 

With that being said, I'm a little nervous about how unwieldy the setup is becoming.